<launch>
  <arg name="bind_ip"          default="127.0.0.1"/> <!-- pilho's IP: 172.22.118.107 -->
  <arg name="dst_ip"           default="127.0.0.1"/> <!-- pilho's IP: 172.22.112.1 -->
  <arg name="ctrl_port"        default="9123"/>
  <arg name="collision_port"   default="9122"/>
  <arg name="status_port"      default="9121"/>
  <arg name="rate_hz"          default="50"/>

  <env name="PYTHONPATH" value="$(find baqu4_udp)/vendor/morai_udp:$(optenv PYTHONPATH)"/>

  <node pkg="baqu4_udp" type="CollisionDataNode.py" name="collision_data_node" required="true" output="screen">
    <param name="ip" value="$(arg bind_ip)"/>
    <param name="port" value="$(arg collision_port)"/>
    <param name="rate_hz" value="$(arg rate_hz)"/>
  </node>

  <node pkg="baqu4_udp" type="VehicleStatusNode.py" name="competition_vehicle_status_node" required="true" output="screen">
    <param name="ip" value="$(arg bind_ip)"/>
    <param name="port" value="$(arg status_port)"/>
    <param name="rate_hz" value="$(arg rate_hz)"/>
  </node>

  <node pkg="baqu4_udp" type="EgoCtrlCmdNode.py" name="ego_ctrl_cmd_node" required="true" output="screen">
    <param name="ip" value="$(arg dst_ip)"/>
    <param name="port" value="$(arg ctrl_port)"/>
    <param name="rate_hz" value="$(arg rate_hz)"/>
    <param name="cmd_topic" value="/ctrl_cmd"/>
    <param name="steer_is_radian" value="true" type="bool"/>
    <param name="max_steer_deg" value="40.0"/>
  </node>

</launch>